packages/reactive_ble_mobile/ios/Classes/BleData extras/CharacteristicInstance.swift

```diff
@@ -0,0 +1,99 @@
+import class CoreBluetooth.CBUUID
+import class CoreBluetooth.CBCharacteristic
+import class CoreBluetooth.CBService
+
+struct CharacteristicInstance: Equatable {
+
+    let id: CharacteristicID
+    let instanceID: CharacteristicInstanceID
+    let serviceID: ServiceID
+    let serviceInstanceID: ServiceInstanceID
+    let peripheralID: PeripheralID
+}
+
+extension CharacteristicInstance {
+    
+    init(_ characteristic: CBCharacteristic) throws {
+        guard let service = characteristic.service
+        else {
+            throw Failure.serviceNotFound
+        }
+        
+        guard let peripheral = service.peripheral
+        else {
+            throw Failure.peripheralNotFound
+        }
+        
+        guard
+            let characteristicIndex = service.characteristics?.filter({ c in c.uuid == characteristic.uuid }).index(of: characteristic)
+        else {
+            throw Failure.characteristicNotFound
+        }
+        
+        guard
+            let serviceIndex = peripheral.services?.filter({ s in s.uuid == service.uuid }).index(of: service)
```

Comment:
If I understand correctly, and also based on our discussions, on iOS we indentify the services by its order in the list of services for some specific characteristic uuid.

Should it get some comment to explain that for the serviceIndex, or is this information useless in the point of view of the public interface?

Also, confusing that you need to filter the `peripheral.services`, since the peripheral comes from `service.peripheral` in the first place

Reply:
It's an implementation detail of the Swift code that the index is used.

I'm not sure what is confusing about filtering `peripheral.services`. It contains services of all kinds and I'm interested in the index within the services of the same kind (hoping that number is stable).